### PR TITLE
docs: :bulb: example python notebooks using `hra-api-client`

### PR DESCRIPTION
**Pull request by bherr2**
Add example notebooks showcasing how to use the HRA-API from Python via the [hra-api-client](https://pypi.org/project/hra-api-client/) module.

### CODE REVIEW

**Comment by andreasbueckle** [comment]
Thanks. I started investigating the notebooks at https://github.com/hubmapconsortium/ccf-ui/blob/main/ccf-api-usage.ipynb and https://github.com/hubmapconsortium/ccf-ui/blob/main/ccf-api-usecase.ipynb. Need some guidance on documentation to move from `ccf-openapi` to `hra-api-client`.

**Comment by andreasbueckle** [comment]
Update:
I started adding emojis to the INDEX and major headings to indicate which code pieces work fine after I updated them, and which do throw errors. For some reason, I keep getting this error: 
``` {python}
MaxRetryError: HTTPSConnectionPool(host='apps.humanatlas.io', port=443): Max retries exceeded with url: [/api/v1/ontology-term-occurences](https://file+.vscode-resource.vscode-cdn.net/api/v1/ontology-term-occurences)?age=%7B%22min%22%3A%2045.0%2C%20%22max%22%3A%2060.0%7D&bmi=%7B%22min%22%3A%2065.0%2C%20%22max%22%3A%2083.0%7D&cache=true&ontology-terms=http://purl.obolibrary.org/obo/UBERON_0000955&cell-type-terms=http://purl.obolibrary.org/obo/CL_0000000&sex=female&spatial=%7B'x':%2084.0,%20'y':%20146.0,%20'z':%2053.0,%20'radius':%2012.0,%20'target':%20'http://purl.org/ccf/latest/ccf.owl (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)')))
```

**Comment by andreasbueckle** [comment]
Update: 
I get this error when running `Run a SPARQL query (POST)`:
```
Exception when calling DefaultApi->sparql_post: (405)
Reason: Method Not Allowed
HTTP response headers: HTTPHeaderDict({'Content-Type': 'text/html; charset=utf-8', 'Content-Length': '22', 'Connection': 'keep-alive', 'cache-control': 'no-cache', 'content-security-policy': "base-uri 'self' cdn.humanatlas.io cdn.jsdelivr.net;script-src 'self' 'unsafe-inline' 'unsafe-eval' cdn.humanatlas.io cdn.jsdelivr.net unpkg.com [www.googletagmanager.com;img-src](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/abueckle/Documents/GitHub/hra-api/notebooks/www.googletagmanager.com;img-src) 'self' 'unsafe-eval' cdn.humanatlas.io cdn.jsdelivr.net unpkg.com [www.googletagmanager.com;connect-src](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/abueckle/Documents/GitHub/hra-api/notebooks/www.googletagmanager.com;connect-src) *;default-src 'self';font-src 'self' https: data:;form-action 'self';frame-ancestors 'self';object-src 'none';script-src-attr 'none';style-src 'self' https: 'unsafe-inline';upgrade-insecure-requests", 'cross-origin-opener-policy': 'same-origin', 'cross-origin-resource-policy': 'same-origin', 'date': 'Mon, 08 Jul 2024 13:14:44 GMT', 'etag': 'W/"16-EB1bOAPfMVsBhP8XDhmnvcfBZ4w"', 'origin-agent-cluster': '?1', 'referrer-policy': 'no-referrer', 'strict-transport-security': 'max-age=15552000; includeSubDomains', 'x-content-type-options': 'nosniff', 'x-dns-prefetch-control': 'off', 'x-download-options': 'noopen', 'x-frame-options': 'SAMEORIGIN', 'x-permitted-cross-domain-policies': 'none', 'x-xss-protection': '0', 'x-envoy-upstream-service-time': '3', 'server': 'envoy', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 428a2ed921cd3013591e242ee4178786.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'ORD51-C4', 'Alt-Svc': 'h3=":443"; ma=86400', 'X-Amz-Cf-Id': 'LB8nvIqLxX1QEswjktkOOvacMelYQN9rqjC7JN_dziFeJS4mNcniEg==', 'Vary': 'Origin'})
HTTP response body: Unsupported operation.
```

We get a similar error when trying out the API endpoint at https://apps.humanatlas.io/api/#post-/v1/sparql: 
image


Similar error when running 'Given a SpatialEntity already placed relative to a reference SpatialEntity retrieve a new direct SpatialPlacement to the given SpatialEntity IRI`:
```
Exception when calling DefaultApi->get_spatial_placement: (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Content-Type': 'application/json; charset=utf-8', 'Content-Length': '72', 'Connection': 'keep-alive', 'cache-control': 'no-cache', 'content-security-policy': "base-uri 'self' cdn.humanatlas.io cdn.jsdelivr.net;script-src 'self' 'unsafe-inline' 'unsafe-eval' cdn.humanatlas.io cdn.jsdelivr.net unpkg.com [www.googletagmanager.com;img-src](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/abueckle/Documents/GitHub/hra-api/notebooks/www.googletagmanager.com;img-src) 'self' 'unsafe-eval' cdn.humanatlas.io cdn.jsdelivr.net unpkg.com [www.googletagmanager.com;connect-src](https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/abueckle/Documents/GitHub/hra-api/notebooks/www.googletagmanager.com;connect-src) *;default-src 'self';font-src 'self' https: data:;form-action 'self';frame-ancestors 'self';object-src 'none';script-src-attr 'none';style-src 'self' https: 'unsafe-inline';upgrade-insecure-requests", 'cross-origin-opener-policy': 'same-origin', 'cross-origin-resource-policy': 'same-origin', 'date': 'Mon, 08 Jul 2024 13:17:22 GMT', 'etag': 'W/"48-LlldxmjVeDK3aB9pU7mbZaega1c"', 'origin-agent-cluster': '?1', 'referrer-policy': 'no-referrer', 'strict-transport-security': 'max-age=15552000; includeSubDomains', 'x-content-type-options': 'nosniff', 'x-dns-prefetch-control': 'off', 'x-download-options': 'noopen', 'x-frame-options': 'SAMEORIGIN', 'x-permitted-cross-domain-policies': 'none', 'x-xss-protection': '0', 'x-envoy-upstream-service-time': '4', 'server': 'envoy', 'X-Cache': 'Error from cloudfront', 'Via': '1.1 428a2ed921cd3013591e242ee4178786.cloudfront.net (CloudFront)', 'X-Amz-Cf-Pop': 'ORD51-C4', 'Alt-Svc': 'h3=":443"; ma=86400', 'X-Amz-Cf-Id': 'jO51qAI6E3dlU9tHboCzU6j36djtTsw7OrxuXkjr1SBom_OmocEdFg==', 'Vary': 'Origin'})
HTTP response body: error='Placement path not found from rui_location to targetIri'
```

**Comment by andreasbueckle** [comment]
Update: I finished going through https://github.com/x-atlas-consortia/hra-api/blob/python-notebooks/notebooks/hra-api-client-usage.ipynb and documented issues in https://github.com/x-atlas-consortia/hra-api/blob/python-notebooks/notebooks/README.md.

**Comment by andreasbueckle** [comment]
I went through https://github.com/x-atlas-consortia/hra-api/blob/python-notebooks/notebooks/hra-api-client-usecase.ipynb now as well and documented issues in https://github.com/x-atlas-consortia/hra-api/blob/python-notebooks/notebooks/README.md

File: src/server/routes/sparql.js [hunk 1]
```diff
@@ -68,6 +68,6 @@ const sparql = async (req, res, _next) => {
   res.end();
 };
 
-const routes = Router().get('/', sparql).post('/', sparql);
+const routes = Router().use('/', sparql).post('/', sparql);
```
Comment (axdanbol) [inline src/server/routes/sparql.js:71]: I think we can remove the `post(...)` here since the `get` was changed into a `use`

**Comment by andreasbueckle** [comment]
I re-opened this branch. @axdanbol is doing some more fixing, then I'll finalize the notebooks.

**Pull request merged**